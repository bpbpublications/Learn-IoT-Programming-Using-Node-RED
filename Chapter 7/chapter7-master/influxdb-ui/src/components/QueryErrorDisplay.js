//  React
import React, { Component } from 'react';

class QueryErrorDisplay extends Component {  

  render() {
    //  If we don't have an error, don't show this
    if(this.props.haserror === false) {
        return null;
    }

    //  Otherwise, display our error
    return (
        <div className="alert alert-warning" role="alert">
            Error: {this.props.error}
        </div>
    );
  }

}

export default QueryErrorDisplay;