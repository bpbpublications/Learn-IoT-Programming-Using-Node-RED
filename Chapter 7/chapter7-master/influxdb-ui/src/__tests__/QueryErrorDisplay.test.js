import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

//  Components under test
import QueryErrorDisplay from '../components/QueryErrorDisplay';

it('Displays no error as expected', () => {
    const tree = renderer.create(
        <QueryErrorDisplay />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

it('Displays the given error information', () => {
    const tree = renderer.create(
        <QueryErrorDisplay haserror='true' error="Some error text" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

