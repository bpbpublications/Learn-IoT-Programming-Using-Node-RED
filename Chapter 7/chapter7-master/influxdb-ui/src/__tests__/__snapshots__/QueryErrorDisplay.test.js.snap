// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Displays no error as expected 1`] = `
<div
  className="alert alert-warning"
  role="alert"
>
  Error: 
</div>
`;

exports[`Displays the given error information 1`] = `
<div
  className="alert alert-warning"
  role="alert"
>
  Error: 
  Some error text
</div>
`;
